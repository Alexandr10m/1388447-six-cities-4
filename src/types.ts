export interface LocalOffer {
  isPremium: boolean;
  pictures: string[];
  price: number;
  isFavourite: boolean;
  grade: number;
  title: string;
  type: string;
  bedroom: number;
  maxAdults: number;
  facilities: string[];
  coords: number[];
  locationZoom: number;
  id: number;
  description: string;
  previewImage: string;
  host: Host;
}

export interface CityOffers {
  city: string;
  cityCoords: number[];
  cityZoom: number;
  localOffers: LocalOffer[] | [];
}

export interface Host {
  id: number;
  name: string;
  isPro: boolean;
  avatarUrl: string;
}

export interface author {
  avatarUrl: string;
  email: string;
  id: number;
  isPro: boolean;
  name: string;
}

export interface Review {
  text: string;
  date: InstanceType<typeof Date>;
  id: number;
  grade: number;
  user: author;
}
