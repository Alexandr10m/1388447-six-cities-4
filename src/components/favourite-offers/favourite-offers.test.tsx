import * as React from "react";
import * as renderer from "react-test-renderer";
import FavouriteOffers from "./favourite-offers";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducer/name-space";
import {BrowserRouter} from "react-router-dom";
import {LocalOffer, CityOffers} from "../../types";


const mockStore = configureStore([]);

const offerWithPremium: LocalOffer = {
  bedroom: 2,
  coords: [48.865610000000004, 2.350499],
  description: `in city center.`,
  facilities: [`Air conditioning`, `Breakfast`],
  grade: 3.6,
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    id: 25,
    isPro: true,
    name: `Angelina`,
  },
  id: 1,
  isFavourite: false,
  isPremium: true,
  locationZoom: 16,
  maxAdults: 8,
  pictures: [`https://`, `https`],
  previewImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
  price: 397,
  title: `Penthouse`,
  type: `hotel`,
};
const offerWithFavourite: LocalOffer = {
  bedroom: 2,
  coords: [48.865610000000004, 2.350499],
  description: `in city center.`,
  facilities: [`Air conditioning`, `Breakfast`],
  grade: 3.6,
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    id: 25,
    isPro: true,
    name: `Angelina`,
  },
  id: 1,
  isFavourite: true,
  isPremium: false,
  locationZoom: 16,
  maxAdults: 8,
  pictures: [`https://`, `https`],
  previewImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
  price: 397,
  title: `Penthouse`,
  type: `hotel`,
};

const favouriteOffers: CityOffers[] = [
  {
    city: `Paris`,
    cityCoords: [48.85661, 2.351499],
    cityZoom: 13,
    localOffers: [offerWithFavourite],
  },
  {
    city: `Amsterdam`,
    cityCoords: [52.38333, 4.9],
    cityZoom: 13,
    localOffers: [offerWithPremium],
  }
];

describe(`Snapshot of FavouriteOffers`, () => {
  it(`Should render correctly`, () => {
    const store = mockStore({
      [NameSpace.STATE]: {},
      [NameSpace.DATA]: {},
      [NameSpace.USER]: {},
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <FavouriteOffers
                favouriteOffers={favouriteOffers}
              />
            </BrowserRouter>
          </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
