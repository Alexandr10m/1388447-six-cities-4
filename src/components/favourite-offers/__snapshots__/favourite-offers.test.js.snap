// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of FavouriteOffers Should render correctly 1`] = `
<section
  className="favorites"
>
  <h1
    className="favorites__title"
  >
    Saved listing
  </h1>
  <ul
    className="favorites__list"
  >
    <li
      className="favorites__locations-items"
    >
      <div
        className="favorites__locations locations locations--current"
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="/Paris"
            onClick={[Function]}
          >
            <span>
              Paris
            </span>
          </a>
        </div>
      </div>
      <div
        className="favorites__places"
      >
        <article
          className="favorites__card place-card"
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="favorites__image-wrapper place-card__image-wrapper"
          >
            <a
              href="/offer/2"
              onClick={[Function]}
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="110"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg"
                width="150"
              />
            </a>
          </div>
          <div
            className="favorites__card-info place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  397
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button place-card__bookmark-button--active button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "72%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/2"
                onClick={[Function]}
              >
                Penthouse, 4-5 rooms + 5 balconies
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              hotel
            </p>
          </div>
        </article>
      </div>
    </li>
    <li
      className="favorites__locations-items"
    >
      <div
        className="favorites__locations locations locations--current"
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="/Amsterdam"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </div>
      </div>
      <div
        className="favorites__places"
      >
        <article
          className="favorites__card place-card"
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="favorites__image-wrapper place-card__image-wrapper"
          >
            <a
              href="/offer/1"
              onClick={[Function]}
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="110"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg"
                width="150"
              />
            </a>
          </div>
          <div
            className="favorites__card-info place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  397
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button  button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "72%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/1"
                onClick={[Function]}
              >
                Penthouse, 4-5 rooms + 5 balconies
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              hotel
            </p>
          </div>
        </article>
      </div>
    </li>
  </ul>
</section>
`;
